src/variant_type.rs

```diff
@@ -208,6 +212,47 @@ impl ToOwned for VariantTy {
     }
 }
 
+impl StaticType for VariantTy {
+    fn static_type() -> Type {
+        unsafe { from_glib(glib_ffi::g_variant_type_get_gtype()) }
+    }
+}
+
+impl SetValue for VariantTy {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        gobject_ffi::g_value_set_boxed(value.to_glib_none_mut().0, this.to_glib_none().0 as glib_ffi::gpointer)
+    }
+}
+
+impl<'a> FromValueOptional<'a> for &'a VariantTy {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
+        let cvty = gobject_ffi::g_value_dup_boxed(value.to_glib_none().0) as *mut glib_ffi::GVariantType;
+        if cvty.is_null() {
+            None
+        } else {
+            Some(VariantTy::from_ptr(cvty))
+        }
+    }
+}
+
+impl StaticType for VariantType {
+    fn static_type() -> Type {
+        unsafe { from_glib(glib_ffi::g_variant_type_get_gtype()) }
+    }
+}
+
+impl SetValue for VariantType {
+    unsafe fn set_value(value: &mut Value, this: &Self) {
+        gobject_ffi::g_value_set_boxed(value.to_glib_none_mut().0, this.to_glib_none().0 as glib_ffi::gpointer)
+    }
+}
+
+impl<'a> FromValueOptional<'a> for VariantType {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
+        Option::<VariantType>::from_glib_full(gobject_ffi::g_value_dup_boxed(value.to_glib_none().0) as *mut glib_ffi::GVariantType)
```

Comment:
This should be `from_glib_none()` and `get_boxed()`.

See also this comment in the code from `from_glib_full`:

```rust
impl FromGlibPtrFull<*const glib_ffi::GVariantType> for VariantType {
    unsafe fn from_glib_full(ptr: *const glib_ffi::GVariantType) -> VariantType {
        // Don't assume ownership of a const pointer.
        // A transfer: full annotation on a `const GVariantType*` is likely a bug.
        VariantTy::from_ptr(ptr).to_owned()
    }
}
```